### PR TITLE
lf hitag hts restore & 82xx improvements

### PR DESCRIPTION
**Pull request by Nya0**

### CODE REVIEW

File: client/src/cmdlfhitaghts.c [hunk 1]
```diff
@@ -454,6 +456,7 @@ static int CmdLFHitagSDump(const char *Cmd) {
                   "    - default key 4F4E4D494B52 (ONMIKR)\n\n"
                   "  8268/8310 password mode: \n"
                   "    - default password BBDD3399\n",
+                  "lf hitag hts dump --82xx -k              -> use def pwd\n"
```
Comment (iceman1001) [inline client/src/cmdlfhitaghts.c:459]: remove the empty -k param.

File: client/src/cmdlfhitaghts.c [hunk 2]
```diff
@@ -534,6 +537,225 @@
...
+
+    lf_hts_read_response_t *config = (lf_hts_read_response_t *)resp.data.asBytes;
+    hitags_config_t tag_config = config->config_page.s;
+    
+    const int hts_mem_sizes[] = {1, 8, 64, 64};
+    int mem_size = hts_mem_sizes[tag_config.MEMT] * HITAGS_PAGE_SIZE;
+
+    if (bytes_read != mem_size) {
+        free(dump);
+        PrintAndLogEx(FAILED, "Wrong length of dump file. Expected %zu bytes, got %zu", mem_size, bytes_read);
```
[the remark this answers is not in the document]
Comment (iceman1001) [reply client/src/cmdlfhitaghts.c:624]: fix string formatter

File: client/src/cmdlfhitaghts.c [hunk 3]
```diff
@@ -534,6 +537,225 @@
...
+            break;
+            case 3:  // crypto mode  
+                if (packet.cmd == HTSF_KEY) {
+                    
+                    if (memcmp(packet.key, &dump_bytes[offset - HITAGS_PAGE_SIZE], HITAGS_PAGE_SIZE * 2) == 0) {
+                        break;
+                    }
+                    auth_changed = true;
+
+                    memcpy(packet.key, &dump_bytes[offset - HITAGS_PAGE_SIZE], HITAGS_PAGE_SIZE * 2);
```
[the remark this answers is not in the document]
Comment (iceman1001) [reply client/src/cmdlfhitaghts.c:699]: verify keylen from input before accessing key array

**Comment by iceman1001** [comment]
Some comments to fix  but nice work!

**Comment by Nya0** [comment]
done!

**Comment by iceman1001** [comment]
Excellent!

**Pull request merged**